File: src/tree_parity_machine/layer/NetLayer.java
package tree_parity_machine.layer;

import tree_parity_machine.neuron.Neuron;

import java.util.Arrays;

public abstract class NetLayer {

    protected int outputs;
    protected int inputs;
    protected Neuron[] neurons;

    @Override
    public String toString() {
        return "NetLayer{" +
                "neurons=" + Arrays.toString(neurons) +
                ", outputs=" + outputs +
                ", inputs=" + inputs +
                '}';
    }
}

File: src/main/kotlin/com/jvvladimir/machine/tree_machine/layer/NetLayer.kt
package com.jvvladimir.machine.tree_machine.layer

import com.jvvladimir.machine.learning.LearningParadigm
import com.jvvladimir.machine.learning.LearningParadigm.HEBBIAN
import com.jvvladimir.machine.tree_machine.neuron.Neuron

abstract class NetLayer {
    protected var outputs = 0
    protected var inputs = 0
    lateinit var neurons: Array<Neuron>
    var paradigm: LearningParadigm = HEBBIAN

    override fun toString() = "NetLayer{ neurons= ${neurons.contentToString()} outputs= $outputs inputs= $inputs}"
}